[valerietanhx] iP

valerietanhx opened the pull request:
# Duke the bunny is here! ૮ ˶ᵔ ᵕ ᵔ˶ ა

> “I don’t know why one can’t chase two bunnies at the same time, even in the literal sense of those words. If you have the hounds, go ahead and pursue.” — Anton Chekhov ([source](https://listcaboodle.com/rabbit-quotes-and-captions/))

Hello! I'm Duke, a bunny here to help you keep track of all the things a busy human needs to do 🐰
Here are all the things I can do for you:

- Remember your things to do
- Remember your deadlines and their due dates
- Remember your events and their timings
- Check off things you've done
- Be _very_ cute while doing all of that!

Look at me holding up this board in the `Ui` class! I work hard to keep all your tasks organised.

```java
public void showBunny() {
        System.out.println("(\\ (\\\n" +
                            "(„• ֊ •„) ♡\n" +
                            "━O━O━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━");
    }
```

And I'm learning to do some other things as well:

- [ ] Find tasks based on their description
- [ ] Remind you of upcoming deadlines and events
- [ ] Be _**even cuter**_!

Please adopt me and give me a loving home!

1. Download my home from [here](https://github.com/valerietanhx/ip/releases/tag/A-Jar). I live in a little JAR!
2. Open my JAR with `java -jar duke.jar` and say hello to me!
3. Add your tasks one by one

I'm so excited to meet you! See you soon 🎀

Commit:
Add Gradle support

valerietanhx committed:
Add Level-1

valerietanhx committed:
Add Level-2

valerietanhx committed:
Add Level-3

valerietanhx committed:
Add Level-4

valerietanhx committed:
Update Level-4

valerietanhx committed:
Update Level-4

valerietanhx committed:
Add A-TextUiTesting

valerietanhx committed:
Refactor code

valerietanhx committed:
Add Level-5

valerietanhx committed:
Add Level-6

valerietanhx committed:
Add adapted code attribution in line with reuse policy

valerietanhx committed:
Level-7

valerietanhx committed:
Add Level-8

valerietanhx committed:
Merge branch 'branch-Level-8'

valerietanhx committed:
Refactor code

valerietanhx committed:
Update Level-7

valerietanhx committed:
Change Duke to a bunny

valerietanhx committed:
Add A-MoreOOP

valerietanhx committed:
Clean up Parser

valerietanhx committed:
Add A-Packages

valerietanhx committed:
Merge remote-tracking branch 'origin/add-gradle-support'

valerietanhx committed:
Add A-JUnit

valerietanhx committed:
Update build.gradle Shadow plugin details

valerietanhx committed:
Add A-CheckStyle

valerietanhx committed:
Add A-JavaDoc

valerietanhx committed:
Add A-CodingStandard

valerietanhx committed:
Add Level-9

valerietanhx committed:
Update Level-9

valerietanhx committed:
Fix merge conflicts

valerietanhx committed:
Fix merge conflicts

valerietanhx committed:
Edit typo in Event class

angjunkang commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,28 @@
+        this.keyword = keyword;
+    }
+
+    /**
```
Perhaps consider using @inheritDoc if you have the same Javadoc for the parent and child method.

angjunkang commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,87 @@
+ */
+public class Parser {
+
+    public static Command parse(String line) throws DukeException {
```
Perhaps the usage of enums here would make it more intuitive?

angjunkang commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,53 @@
+        return isDone;
+    }
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
Maybe comments can be placed directly above the code and indented relative to their position in the code?

angjunkang commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    /**
+     * Marks a task as undone.
```
Should this be unmark instead of mark?

angjunkang reviewed (commented):
Overall, I found your code easy to read, clear and concise. There were only a few minor code quality issues like comments and usage of enums. LGTM!

NBQian commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,28 @@
+import duke.ui.Ui;
+
+/**
+ * A command that finds tasks containing a given keword.
```
possible typo: did you mean "keyword" ?

NBQian commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,87 @@
+    public static Command parse(String line) throws DukeException {
+        String[] chunked = line.split(" ");
+        String command = chunked[0];
+        if (command.equals("bye")) {
```
maybe you can consider using switch statements instead of multiple if else statements with repeated calling of command.equals().

NBQian commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + String.format(" (by: %s %s)", dateFormatter.format(datetime), timeFormatter.format(datetime));
```
this line seems a little bit too long, maybe consider dividing it to two lines

NBQian commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,147 @@
+    private final Scanner sc = new Scanner(System.in);
+    private boolean firstTime = false;
+
+    public void setFirstTime(boolean b) {
```
maybe add some javadoc to explain this method?

NBQian reviewed (commented):
LGTM! just some minor suggestions.

valerietanhx committed:
Fix bug in find command

valerietanhx committed:
Add Level-10

valerietanhx committed:
Add A-Assertions

valerietanhx committed:
Improve code quality

valerietanhx committed:
Merge pull request #2 from valerietanhx/branch-A-Assertions

Add A-Assertions

valerietanhx committed:
Merge branch 'master' into branch-A-CodeQuality

valerietanhx committed:
Merge pull request #3 from valerietanhx/branch-A-CodeQuality

Improve code quality

valerietanhx committed:
Reduce line length

The toString method of Deadline.java has a line length of 136 characters.

Line length should be reduced to improve readability.

Let's insert a line break at an appropriate place to reduce the line length to <120 characters.

valerietanhx committed:
Add description editing functionality

Users are unable to edit a task after it has been added. Instead, they have to manually delete and add a new one.

Task editing functionality should be added to enhance user experience. Users would be able to rectify any errors they have made when adding the task.

Let’s add the ability to edit task descriptions.

valerietanhx committed:
Lengthen bunny divider

valerietanhx committed:
Delete unnecessary file

valerietanhx committed:
Make display picture a circle

valerietanhx committed:
Improve GUI

valerietanhx committed:
Add product name to GUI

valerietanhx committed:
Add product picture

valerietanhx committed:
Add user guide

Add user guide in README.md

valerietanhx committed:
Edit typo in user guide
